creusot/src/backend/logic.rs

```diff
@@ -172,12 +173,16 @@ fn body_module<'tcx>(ctx: &mut Why3Generator<'tcx>, def_id: DefId) -> (Module, C
             decls.push(val_decl(ctx, &mut names, def_id));
         } else if body.is_pure() {
             let def_sig = sig.clone();
-            let val = util::item_type(ctx.tcx, def_id).val(sig.clone());
+            let val = ity.val(sig.clone());
             decls.push(Decl::ValDecl(val));
             decls.push(val_decl(ctx, &mut names, def_id));
-            decls.push(Decl::Axiom(definition_axiom(&def_sig, body)));
+            if sig.trigger.is_none() {
```

Comment:
I'm pretty sure this means it is applied even if `simple_triggers` isn't turned on, no?

Reply:
I don't like this test, is there a way to make it more obvious what would cause this to apply? ie: perhaps `sig.uses_simple_triggers()`?